django/contrib/auth/tokens.py

```diff
@@ -12,20 +15,42 @@ class PasswordResetTokenGenerator:
     """
     key_salt = "django.contrib.auth.tokens.PasswordResetTokenGenerator"
     algorithm = None
-    secret = None
+    _secret = None
+    secrets = None
 
     def __init__(self):
-        self.secret = self.secret or settings.SECRET_KEY
         # RemovedInDjango40Warning: when the deprecation ends, replace with:
         # self.algorithm = self.algorithm or 'sha256'
         self.algorithm = self.algorithm or settings.DEFAULT_HASHING_ALGORITHM
 
+    # This is WIP. Needs some tests at least.
+    # Idea is needing to allow the setter on .secret, whilst lazily reading from
+    # SECRET_KEYS.
+    @property
+    def secret(self):
+        return self._secret or self._get_secrets()[0]
```

Comment:
It is not exactly clear to me why this would need to be a property instead of calling doing:
```
self.secret = self.secret or self._getsecrets()[0]
```
I see that you might get into some troubles in `_get_secret` then. But this begs the question: We do not document the `secret` attribute at all (at least I didn't see it), so we could just remove that in favor of `secrets`. If we want to be nice a deprecation path could look like this:
```
def __init__(self):
  if self.secret:
    self.secrets = [self.secret]
    warnings.warn("Using deprecated feature set, blablabla")
  self.secrets = self.secrets or settings.SECRET_KEYS
```
Then you can simply drop the properties and `get_secrets` all together.